evm/utils/blobs.py

```diff
@@ -60,5 +65,63 @@ def calc_chunk_root(collation_body: bytes) -> bytes:
             COLLATION_SIZE
         ))
 
-    chunks = chunk_iterator(collation_body)
+    chunks = iterate_chunks(collation_body)
     return calc_merkle_root(chunks)
+
+
+def check_body_size(body):
+    if len(body) > COLLATION_SIZE:
+        raise ValueError("Collation body exceeds maximum allowed size")
+    return body
+
+
+@apply_to_return_value(check_body_size)
+@apply_to_return_value(lambda v: zpad_right(v, COLLATION_SIZE))
+@apply_to_return_value(b"".join)
+def serialize_blobs(blobs: Iterable[bytes]) -> Iterator[bytes]:
+    """Serialize a sequence of blobs and return a collation body."""
+    for blob in blobs:
+        if len(blob) == 0:
+            raise ValueError("Cannot serialize blob of length 0")
+        if len(blob) > 31 * 2**((COLLATION_SIZE - 1).bit_length() - 5):
+            raise ValueError("Cannot serialize blob of size {}".format(len(blob)))
+
+        blob_index = 0
```

Comment:
I assume initializing this value to zero here is for the case where `len(blob) < 31` which causes the loop below to not run, thus `blob_index` never gets initialized. Maybe we could handle this case explicitely instead?